docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+  <br>
+  Creating user-defined-networks using NADs may introduce unexpected behaviour and collisions with NAD object the controller manage.
+- Managed NAD object should not be deleted manually, in order to delete a network, the corresponding CR instance should be deleted.
+- Only one primary network per namespace is supported.
+- Make sure no workloads exist on a namespace before creating primary network in that namespace.
+- For the cluster-scoped CRD, its recommended to use `kubernetes.io/metadata.name` label to specify the target namespaces.
+
+#### UDN Functional Implementation Details
+
+OVN offers the ability to create multiple virtual topologies. As with secondary networks in OVN-Kubernetes today,
+separate topologies are created whenever a new network is needed. The same methodology will be leveraged for this design.
+Whenever a new network of type layer 3 or layer 2 is requested, a new topology will be created for that network where
+pods may connect to.
+
+The limitation today with secondary networks is that there is only support for east/west traffic. This RFE will address
+adding support for user-defined primary and secondary network north/south support. In order to support north/south
+traffic, pods on different networks need to be able to egress, typically using the host’s IP. Today in shared gateway
+mode we use a Gateway Router (GR) in order to provide this external connectivity, while in local gateway mode, the host
+kernel handles SNAT’ing and routing out egress traffic. Ingress traffic also follows similar, and reverse paths. There
+are some exceptions to these rules:
+
+1. MEG traffic always uses the GR to send and receive traffic.
+2. Egress IP on the primary NIC always uses the GR, even in local gateway mode.
+3. Egress Services always use the host kernel for egress routing.
+
+To provide an ingress/egress point for pods on different networks the most simple solution may appear to be to connect
+them all to a single gateway router. This introduces an issue where now networks are all connected to a single router,
+and there may be routing happening between networks that were supposed to be isolated from one another. Furthermore in
+the future, we will want to extend these networks beyond the cluster, and to do that in OVN would require making a
+single router VRF aware, which adds more complexity into OVN.
+
+The proposal here is to create a GR per network. With this topology, OVN will create a patch port per network to the
+br-ex bridge. OVN-Kubernetes will be responsible for being VRF/network aware and forwarding packets via flows in br-ex
+to the right GR. Each per-network GR will only have load balancers configured on it for its network, and only be able to
+route to pods in its network. The logical topology would look something like this, if we use an example of having a
+cluster default primary network, a layer 3 primary network, and a layer 2 primary network:
+
+![VRF Topology](../images/VRFs.svg)
```

Comment:
In the preview I see this weirdness for all pictures:
![Screenshot from 2025-05-05 09-42-20](https://github.com/user-attachments/assets/ac253019-1b3e-48ca-a8e4-0c23053aa143)
can you check our documentation CI lane - the md files you can download from artifacts to check if these images are showing up correctly on the website.. the CI lane is called "Test and Deploy static content to Pages" -> it has preview pages

Reply:
i'll look, but if in github if you click the ... then view file on tthe okep-5193-user-defined-networks.md it will render the pictures in the document:

https://github.com/ovn-kubernetes/ovn-kubernetes/blob/2d1f272abcdb3143e43e30b7a726bf1748e31e71/docs/okeps/okep-5193-user-defined-networks.md

Reply:
confirmed i see the pictures in the output from the CI job:

![image](https://github.com/user-attachments/assets/034b6b7b-3b1d-4655-a0ac-4d02b290582a)